Quotient/syncdata.cpp

```diff
@@ -216,10 +218,15 @@ void SyncData::parseJson(const QJsonObject& json, const QString& baseDir)
         roomData.reserve(roomData.size() + static_cast<size_t>(rs.size()));
         for (auto roomIt = rs.begin(); roomIt != rs.end(); ++roomIt) {
             QJsonObject roomJson;
-            if (!baseDir.isEmpty()) {
-                // Loading data from the local cache, with room objects saved in
-                // individual files rather than inline
-                roomJson = loadJson(baseDir + fileNameForRoom(roomIt.key()));
+            // Normally (i.e. in a /sync response) the received JSON is
+            // self-contained; but the local cache stores state for each room in
+            // its own file, loaded below
+            if (Q_UNLIKELY(!baseDir.isEmpty())) {
+                roomJson = loadJson(
+                    baseDir
+                    + (roomIt->isUndefined() // lib pre-0.8.1.2 = cache pre-11.3
```

Comment:
This should have been `roomIt->isNull()` or even `!roomIt->isObject()`... because null value is a defined value, so this condition will always be false for old caches.